src/layer/S3SLSTRLayer.ts

```diff
@@ -49,4 +49,26 @@ export class S3SLSTRLayer extends AbstractSentinelHubV3Layer {
       hasMore: response.data.hasMore,
     };
   }
+
+  public async findDates(
+    bbox: BBox,
+    fromTime: Date,
+    toTime: Date,
+    datasetSpecificParameters: {
+      maxCloudCoverage?: number;
+      orbitDirection: OrbitDirection | null;
+      view: 'NADIR' | 'OBLIQUE';
+    } = { orbitDirection: OrbitDirection.DESCENDING, view: 'NADIR' },
+  ): Promise<Date[]> {
+    const findDatesDatasetParameters: S3SLSTRFindTilesDatasetParameters = {
+      type: this.dataset.datasetParametersType,
+      orbitDirection: datasetSpecificParameters.orbitDirection,
+      view: datasetSpecificParameters.view,
+    };
+
+    return super.findDates(bbox, fromTime, toTime, {
+      maxCloudCoverage: datasetSpecificParameters && datasetSpecificParameters.maxCloudCoverage,
```

Comment:
Note that if you use this:
```
const x = datasetSpecificParameters && datasetSpecificParameters.maxCloudCoverage;
```

then the value of `x` is either `datasetSpecificParameters.maxCloudCoverage` or `false` - which is probably not we want in this case?

Reply:
Thanks, it was wrong. I think it's not applicable here anymore, but would appreciate a check in the `Abstract...Layer` classes

Reply:
I didn't notice it anymore, resolving.